### DISCUSSION TITLE
How do I take a screenshot?

### DISCUSSION THREAD

**Discussion by BrianPugh** [Textualize/textual#745] [Q&A] [answered]
In the `rich` documentation, it says I can pass the parameter `console = Console(record=True)` and then later invoke `export_svg` and then probably convert it to a png or jpg.

However, [in this commit](https://github.com/Textualize/textual/commit/ffcbe4ba1e04f0bace7d7e8b87aa2e0390555b00), the ability to pass in your own `console` object was removed.

So, my question is, whats the best way to programmatically take a screenshot of the running textual application? Use case: I'm making a dashboard and I want to bind the button "s" so that it saves a copy of the current dashboard state to a timestamped image.

**Comment by willmcgugan** [comment, marked as answer]
There's a method for this in the css branch. For the master branch, try constructing a Console and printing the app.

**Reply by BrianPugh** [reply]
That worked! For anyone who wants the explicit answer:

```
from textual.app import App
from rich.console import Console

class MyApp(App):
    async def on_load(self) -> None:
        await self.bind("s", "screenshot", "Screenshot")

    def action_screenshot(self) -> None:
        console = Console(record=True)
        console.print(self)
        console.save_svg("screenshot.svg", title="MyTitle")
```
